README.md

```diff
@@ -1,24 +1,108 @@
-# README
+# Bookstore Api
+### Development
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+Want to contribute? Great!
+The application requires Ruby on Rails to run.
 
-Things you may want to cover:
+Install the dependencies.
 
-* Ruby version
+```sh
+$ cd bookstore-api
+$ bundle install
+```
 
-* System dependencies
+Create, migrate, seed database and start the server.
 
-* Configuration
+```sh
+$ rails db:create db:migrate db:seed
+$ rails s -b 0.0.0.0 -p 3000
+```
 
-* Database creation
+### Github Integration
 
-* Database initialization
+To integrate this application with github follow the steps below:
 
-* How to run the test suite
+##### Generate a personal access token in Github:
+Access [Github PAT] page.
+Click in Generate new token button.
+Check the repo, and admin:repo_hook permissions.
+Click in Generate token button.
+Create the GITHUB_PERSONAL_ACCESS_TOKEN env with your token.
+Ex:
+```sh
+export GITHUB_PERSONAL_ACCESS_TOKEN="gvdaisdvbakjsbvdkajhb"
+```
 
-* Services (job queues, cache servers, search engines, etc.)
+##### Download ngrok and expose your 3000 port.
 
-* Deployment instructions
+Download ngrok in this link [ngrok].
+Expose the port 3000 executing the following command: 
+```sh
+./ngrok http 3000
+```
+Copy the generated url.  
+```sh
+Forwarding http://2db927ba.ngrok.io -> http://localhost:3000
+```
 
-* ...
+Concatenate the copied url with '/github_webhooks' and create the GITHUB_CALLBACK_URL env var.
+Ex:
+```sh
+export GITHUB_CALLBACK_URL="http://2db927ba.ngrok.io/github_webhooks"
+```
+
+##### Create a GITHUB_REPO env with your issue's repository
+
+Create a new empty repository and point you env to it 
+ex:
+```sh
+export GITHUB_REPO="jotacervan/test-authors-bio".
+```
+*obs: It needs to be a repository that you have admin permissions*
+
+##### Generate a secret to make your integration secure
+
+Generate the secret with rails
+```sh
+rails secret
+>> khsdbflajhsbdfkjahblakdbfnal
+```
+and put it in the GITHUB_WEBHOOK_SECRET env var.
+```sh
+export GITHUB_WEBHOOK_SECRET="khsdbflajhsbdfkjahblakdbfnal"
+```
+
+Execute the Github integration task:
+
+```sh
+rails github:integration
+```
+### Required environment variable's table
+
+| ENV VARS | Description  |
+| ------- | --- |
+| GITHUB_PERSONAL_ACCESS_TOKEN | Personal access token from github |
+| GITHUB_WEBHOOK_SECRET | You can generate a secret with rails secret |
+| GITHUB_CALLBACK_URL | Url generated by the ngrok |
+| GITHUB_REPO | Repository to manage authors and bios |
+
+### Todos
+
+ - Write Tests.
+ - Create user authentication.
+ - Create page to manage books.
+ - Dockerize the application.
+ - Implement a bulk import script.
+ - Implement Redis and Sidekiq to proccess import scripts.
```

Comment:
Why do you think implementing Sidekiq in this project is important?

Reply:
I was thinking about the need of a new publisher to upload all his books, so I would create a bulk import and I would send it to the sidekiq to process the import in the background, assuming that one publisher has a big amount of books.

Reply:
What if you get an unexpected error when processing a webhook? You might end up with inconsistent data between you database and github. Do you see any ways that you could mitigate this problem? Could you list them here please?

Reply:
Sure, processing the webhook queue in the worker is a way to mitigate this problem, I would configure the service to retry, and even if with the retry something goes wrong, I would built an administration area to check the the failed process, and fix it.

Reply:
Cool, thanks =)